pkg/gatewayconfigurationserver/middleware.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gatewayconfigurationserver
+
+import (
+	"strings"
+
+	echo "github.com/labstack/echo/v4"
+	"go.thethings.network/lorawan-stack/pkg/auth/rights"
+	web_errors "go.thethings.network/lorawan-stack/pkg/errors/web"
+	"go.thethings.network/lorawan-stack/pkg/log"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"google.golang.org/grpc/metadata"
+)
+
+func (gcs *GatewayConfigurationServer) handleError() echo.MiddlewareFunc {
```

Comment:
Why do we keep adding new error handling middlewares? The idea was to have only one so that we return consistent responses and don't have to make changes in 10 places.

Reply:
Still open